### PR TITLE
Introduce AFL_EXIT_ON_SEED_ISSUES

### LINKED ISSUE #2067: Introduce ``AFL_EXIT_ON_SEED_ISSUES`` for LibAFL
**Is your feature request related to a problem? Please describe.**
For some of our projects, we generate seeds based on the source code and initialize the fuzzer with them. It would be great to have the possibility to exit on any initial seeds leading to a solution; similar to ``AFL_EXIT_ON_SEED_ISSUES``.

**Describe the solution you'd like**
Modify the ``load_initial_inputs`` API to include a boolean flag to allow solutions and pass them down to ``load_file``
Something like:
``` rust
    fn load_file<E, EM, Z>(
        ..
        allow_if_solution: bool, // here
    ) -> Result<(), Error>
    where
        ...
    {
      ..
        if forced {
          ..
        } else {
            let (res, _) = fuzzer.evaluate_input(self, executor, manager, input.clone())?;
            if res == ExecuteInputResult::Solution && !allow_if_solution { // here
                return Err(Error::corpus_error("Solutions are not allowed in the initial corpus")); // here
            }
        }
        Ok(())
    }
```

**Describe alternatives you've considered**
I could check if there are any solutions after ``state.must_load_initial_inputs`` is done but then I would have to wait for all seeds to be processed. 

**Additional context**
None

### PR DESCRIPTION
**Pull request by R9295**
Implements and closes  #2067

### CODE REVIEW

File: libafl_bolts/src/lib.rs [hunk 1]
```diff
@@ -322,6 +322,8 @@ pub enum Error {
     OsError(io::Error, String, ErrorBacktrace),
     /// Something else happened
     Unknown(String, ErrorBacktrace),
+    // Error with the corpora
+    CorpusError(String, ErrorBacktrace),
```
Comment (domenukk) [inline libafl_bolts/src/lib.rs:326]: I would just name this `Corpus` or `CorpusInvalid` - Error is redundant.
Ideally we would map the error to one of the other errors but I'm not sure if one of them applies (maybe `IllegalState`?)

**Comment by R9295** [comment]
Whoops, I accidentally force pushed.

**Comment by domenukk** [comment]
Can you `git checkout 4676008` and force push that again? :D

File: libafl_bolts/src/lib.rs (hunk 1 above)
Comment (R9295) [reply libafl_bolts/src/lib.rs]: IMO `IllegalState` seems more appropriate for when an invalid action is performed by the user or if external sources modified the items the state depends on (eg. deleting the corpus entry while the fuzzer is running leading to inaccessible corpus entry). Though we could keep it `IllegalState` for simplicity?

File: libafl/src/state/mod.rs [hunk 2]
```diff
@@ -703,6 +708,7 @@ where
         manager: &mut EM,
         forced: bool,
         loader: &mut dyn FnMut(&mut Z, &mut Self, &Path) -> Result<I, Error>,
+        exit_on_solution: bool,
```
Comment (domenukk) [inline libafl/src/state/mod.rs:711]: Can we either add a `LoadConfig` struct with the bools and loader in it (probably pref this one), or move the `loader` up so the booleans are next to each other?

File: libafl/src/state/mod.rs [hunk 3]
```diff
@@ -193,6 +193,19 @@ pub trait HasLastReportTime {
     fn last_report_time_mut(&mut self) -> &mut Option<Duration>;
 }
 
+/// Struct that holds the options for input loading
+pub struct LoadConfig<'a, I, S, Z> {
```
Comment (domenukk) [inline libafl/src/state/mod.rs:197]: Needs to be
`#[cfg(feature = "std")]`

**Comment by domenukk** [comment]
Will merge once CI is green :)

File: libafl_bolts/src/lib.rs (hunk 1 above)
Comment (R9295) [reply libafl_bolts/src/lib.rs]: Yeah, makes more sense.

**Pull request merged**